lib/_http_agent.js

```diff
@@ -65,7 +65,7 @@ function Agent(options) {
         if (self.sockets[name])
           count += self.sockets[name].length;
 
-        if (count >= self.maxSockets || freeLen >= self.maxFreeSockets) {
+        if (count > self.maxSockets || freeLen >= self.maxFreeSockets) {
```

Comment:
`freeLen >= self.maxFreeSockets` is right here.
If `freeLen` equal `maxFreeSockets`, should close the socket instead of push it to `freeSockets` list.